test/language/arguments-object/mapped/mapped-arguments-nonwritable-nonconfigurable-descriptors-4.js

```diff
@@ -0,0 +1,39 @@
+// Copyright (C) 2017 Caio Lima. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+info: Mapped arguments property descriptor change to non-writable and non-configurable
+description: >
+    Mapping stop working when property is set to non-writable. Change the
+    descriptor using [[DefineOwnProperty]] to {writable: false}, change argument[0]
+    value to 2 using [[DefineOwnProperty]] and then
+    change property descriptor to {configurable: false}.
+flags: [noStrict]
+esid: sec-arguments-exotic-objects-defineownproperty-p-desc
+includes: [propertyHelper.js]
+---*/
+
+function argumentsNonWritableThenNonConfigurableWithInterveningDefineOwnPropertyDescriptor(a) {
+  Object.defineProperty(arguments, "0", {writable: false});
+  Object.defineProperty(arguments, "0", {value: 2});
+  Object.defineProperty(arguments, "0", {configurable: false});
```

Comment:
for clarity, this case could be exposed in a more simple test for better clarity:

```js
function fn(a) {
   Object.defineProperty(arguments, "0", {writable: false, enumerable: false});
   Object.defineProperty(arguments, "0", {value: 2});

   assert.sameValue(arguments[0], 2, 'a value can be defined to a non-writable, configurable prop');
   
   arguments[0] = 42;

   console.log(arguments[0], 2, 'a new value cannot be set to a non-writable prop');
}

fn(1);
```

And also split to a new file:

```js
function fn(a) {
   Object.defineProperty(arguments, "0", {writable: false, configurable: false});
   assert.throws(TypeError, function() {
    Object.defineProperty(arguments, "0", {value: 2});
  });
}

fn(1);
```